imgee/tasks.py

```diff
@@ -1,35 +1,83 @@
+import re
 import redis
-from imgee import app
+
+
+class InvalidRedisQueryException(Exception):
+    pass
 
 
 class TaskRegistry(object):
-    def __init__(self, name='default', connection=None):
-        self.connection = redis.from_url(connection) if connection else None
+    def __init__(self, name='default', connection=None, app=None):
+        if app:
+            self.init_app(name, connection)
+
+    def init_app(self, name='default', connection=None):
+        self.connection = connection or self.set_connection_from_url()
         self.name = name
-        self.key = 'imgee:registry:%s' % name
+        self.key_prefix = 'imgee:registry:%s' % name
+        self.filename_pattern = '^[a-z0-9\_\.]+$'
+        self.pipe = self.connection.pipeline()
 
-    def set_connection(self, connection=None):
-        connection = connection or app.config.get('REDIS_URL')
-        self.connection = redis.from_url(connection)
+    def is_valid_query(self, query):
+        return bool(re.match(self.filename_pattern, query))
 
-    def add(self, taskid):
-        self.connection.sadd(self.key, taskid)
+    def set_connection_from_url(self, url=None):
+        from imgee import app
+        url = url or app.config.get('REDIS_URL')
+        self.connection = redis.from_url(url)
+        return self.connection
```

Comment:
Don't return it if you're setting it under self. This entire method's could be inside the `init_app` method.